### DISCUSSION TITLE
Is there a way for the display mode to work with line breaks?

### DISCUSSION THREAD

**Discussion by frnsys** [KaTeX/KaTeX#3277] [Q&A] [answered]
I recall being able to do the following in MathJax:

```
$$
x+2
$$
```

But I noticed that doesn't work in KaTeX. I have to do:

```
$$x+2$$
```

I have several old files that use the former format. Is there a way to allow that in KaTeX? I looked through the options but wasn't able to find anything.

**Comment by edemaine** [comment]
I'm pretty sure both inputs should work fine. Are you using `auto-render`, or something else to trigger KaTeX (e.g. a Markdown plugin)? If the latter, it's probably the plugin at fault. Can you share an example of exactly how you're using it that fails?

**Comment by frnsys** [comment, marked as answer]
Thanks for your quick response. I'm using `auto-render`. This is the snippet I'm using:

``` 
document.addEventListener("DOMContentLoaded", function() {
    renderMathInElement(document.body, {
        delimiters: [
          {left: "$$", right: "$$", display: true},
          {left: "¦", right: "¦", display: false},
        ]
    });
});
```

It is applied to HTML generated from markdown, so the actual HTML it's dealing with looks like:

```
<p>
$$<br>
x+2<br>
$$
</p>
```

**Reply by edemaine** [reply]
The problem is with the `<br>`s, which splits the text elements into `$$`, `x+2`, `$$` respectively. I imagine if you remove them, auto-render will work.

Ideally you can instead use a plugin that calls KaTeX *during the Markdown process* instead of afterwards. Which plugin depends on what Markdown parser you're using.

**Reply by frnsys** [reply]
Great, thank you, I'll look into that.
